xbmc/application/Application.cpp

```diff
@@ -668,6 +668,13 @@ bool CApplication::Initialize()
       }
     }
 
+#if defined(TARGET_ANDROID)
+    if (CServiceBroker::GetSettingsComponent()->GetAdvancedSettings()->m_guiVideoLayoutTransparent)
+    {
+      CLog::Log(LOGINFO, "XBMCApp: VideoLayout view was set to transparent.");
+      CXBMCApp::Get().SetVideoLayoutBackgroundColor(0);
+    }
+#endif
```

Comment:
is there a reason this cant be placed in one of the platform specific init stages, or in windowing?

Reply:
I'm not sure of the best place. I just reused fritsch's code.

Reply:
If anyone can clarify where the Android init stuff is, I can try it out.
Or other suggestions.

Reply:
Maybe move here:

```diff
diff --git a/xbmc/platform/android/activity/XBMCApp.cpp b/xbmc/platform/android/activity/XBMCApp.cpp
index 437c7371d9..64f4dff6ca 100644
--- a/xbmc/platform/android/activity/XBMCApp.cpp
+++ b/xbmc/platform/android/activity/XBMCApp.cpp
@@ -265,6 +265,12 @@ void CXBMCApp::onStart()
         std::make_unique<CJNIActivityManager>(getSystemService(CJNIContext::ACTIVITY_SERVICE));
     m_inputHandler.setDPI(GetDPI());
     runNativeOnUiThread(RegisterDisplayListenerCallback, nullptr);
+
+    if (CServiceBroker::GetSettingsComponent()->GetAdvancedSettings()->m_guiVideoLayoutTransparent)
+    {
+      CLog::Log(LOGINFO, "XBMCApp: VideoLayout view was set to transparent.");
+      SetVideoLayoutBackgroundColor(0);
+    }
   }
 }
```
 
not tested although

Reply:
I've tried moving it but `CServiceBroker::GetSettingsComponent()` ends up being a nullptr at that stage.

Reply:
you can try move it on `CPlatformAndroid` (PlatformAndroid.cpp)
by add a new override to use `InitStageThree` method

Reply:
Thanks. That works. :+1: